src/metabase/sync_database/analyze.clj

```diff
@@ -259,3 +175,11 @@
             (log/info (u/format-color 'blue "%s Analyzed table '%s'." (u/emoji-progress-bar <> tables-count) table-name))))))
 
     (log/info (u/format-color 'blue "Analysis of %s database '%s' completed (%s)." (name driver) (:name database) (u/format-nanoseconds (- (System/nanoTime) start-time-ns))))))
+
+(defn analyze-database
+  "analyze all the tables in one database"
+  [db]
+  (analyze-data-shape-for-tables! (->> db
+                                       :id
```

Comment:
I might be missing something, but my read, this has resulted in some missing behaviour when adding a DB.

Right now, when you add a DB, the `process-sync-database-event` handler is triggered in https://github.com/metabase/metabase/blob/sync-analyze-cache-split/src/metabase/events/sync_database.clj#L22

this calls 'sync-database!' https://github.com/metabase/metabase/blob/sync-analyze-cache-split/src/metabase/sync_database.clj#L68

which ends up in 'sync-database-with-tracking'
https://github.com/metabase/metabase/blob/sync-analyze-cache-split/src/metabase/sync_database.clj#L39

This however, only does an "analysis" unless I am mistaken. 

This seems to break the current sequence for adding a database where 
1. the schema is sync'd
2. the analysis step does the equivalent of the new [cache-values, analyze, classify]
and instead, the value caching and classification only happens when the nightly process finds them.

This is a regression and  not a good outcome for users as new databases aren't really useable when added.

To rectify this I think
1. Classification is only useful when new information appears. Practically, this means a new fingerprint has been generated. Hence, it should be called from the analyze task.
2. When adding a new DB, we should schedule both the chained (analyze + classify) and cache-values tasks for immediate execution.